nvflare/fuel/hci/client/cli.py

```diff
@@ -132,6 +136,17 @@ def __init__(
         if readline:
             readline.set_history_length(cli_history_size)
 
+        signal.signal(signal.SIGUSR1, partial(self.session_signal_handler))
+
+    def session_ended(self, message):
+        self.write_error(message)
+        os.kill(os.getpid(), signal.SIGUSR1)
+
+    def session_signal_handler(self, signum, frame):
+        if self.session_monitor_thread.is_alive():
+            self.session_monitor_thread.join()
+        sys.exit()
```

Comment:
Since the start of session monitor thread is defined in api's "session_monitor" method.

Is it possible to put the join thread logic in api's method?

So that who uses api does not need to worry about this?

Reply:
Good idea. I can move the join logic to the api or I can still return the thread and instead let the user both start and join. Either way works depending on what we want.